src/content/blog/2024-08-05-rafiki-code-architecture.mdx

```diff
@@ -0,0 +1,203 @@
+  >(
+    resourceServerSpec,
+    {
+      path: '/incoming-payments',
+      method: HttpMethod.POST
+    },
+    validatorMiddlewareOptions
+  ),
+  getWalletAddressUrlFromRequestBody,
+  createTokenIntrospectionMiddleware({
+    requestType: AccessType.IncomingPayment,
+    requestAction: RequestAction.Create
+  }),
+  httpsigMiddleware,
+  getWalletAddressForSubresource,
+  incomingPaymentRoutes.create
+)
+```
+
+In this mounting of the `POST /incoming-payments` route, all of the middleware as well as the main handler (`incomingPaymentRoutes.create`) can be seen being attached to that mounting. Note how the `incomingPaymentRoutes` services is acquired from the container using the name that was passed into the function that attached that service to the IoC container. This pattern can be followed for other routes to see which services handle fulfilling requests made to the routes on the Rafiki `backend`.
+
+Finally, the `backend` also launches a GraphQL server that extends an API defined by [this schema](https://github.com/interledger/rafiki/blob/main/packages/backend/src/graphql/schema.graphql).
+
+### The Rafiki Authorization Server
+
+The Authorization Server on Rafiki is structured similarly to the backend, in that it injects services into an IoC container which are retrieved by routes on Koa server to perform the business logic.
+
+[index.ts](https://github.com/interledger/rafiki/blob/main/packages/auth/src/index.ts)
+```
+import { Ioc, IocContract } from '@adonisjs/fold'
+import { createGrantRoutes } from './grant/routes'
+
+...
+
+const container: IocContract<AppServices> = new Ioc()
+
+...
+
+container.singleton('grantRoutes', async (deps: IocContract<AppServices>) => {
+  return createGrantRoutes({
+    grantService: await deps.use('grantService'),
+    clientService: await deps.use('clientService'),
+    accessTokenService: await deps.use('accessTokenService'),
+    accessService: await deps.use('accessService'),
+    interactionService: await deps.use('interactionService'),
+    logger: await deps.use('logger'),
+    config: await deps.use('config')
+  })
+})
+```
+
+Again, note the route service for grants is mounted to the container, and then subsequently referenced when attaching service functions to the relevant routes.
+
+[app.ts](https://github.com/interledger/rafiki/blob/main/packages/auth/src/app.ts)
+```
+const grantRoutes = await this.container.use('grantRoutes')
+
+...
+
+/* Back-channel GNAP Routes */
+// Grant Initiation
+router.post<DefaultState, CreateContext>(
+  '/',
+  createValidatorMiddleware<CreateContext>(openApi.authServerSpec, {
+    path: '/',
+    method: HttpMethod.POST
+  }),
+  grantInitiationHttpsigMiddleware,
+  grantRoutes.create
+)
+```
+
+Like the `backend`, the `auth` server also extends a GraphQL API to perform admin functions. The API is modeled by this [GraphQL schema](https://github.com/interledger/rafiki/blob/main/packages/auth/src/graphql/schema.graphql).
+
+### The Rafiki Frontend
```

Comment:
```suggestion
### Rafiki Admin
```

Reply:
Would "The Rafiki Admin Frontend" work too? Since the package itself is called `frontend` in the repo I thought it might be a good way to help connect those two dots.